plugins/wporg-5ftf/includes/pledge-form.php

```diff
@@ -220,6 +262,86 @@ function render_form_manage() {
 	return ob_get_clean();
 }
 
+/**
+ * Email the pledge admin a temporary link they can use to manage their pledge.
+ *
+ * @param int    $pledge_id
+ * @param string $unverified_admin_email
+ *
+ * @return true|WP_Error
+ */
+function send_manage_pledge_link( $pledge_id, $unverified_admin_email ) {
+	$valid_admin_email = get_post( $pledge_id )->{ PledgeMeta\META_PREFIX . 'org-pledge-email' };
+
+	if ( $valid_admin_email === $unverified_admin_email ) {
+		$subject = __( 'Updating your Pledge', 'wporg-5ftf' );
+		$message =
+			'Howdy heres the link to edit your pledge, ' .
+			Email\get_authentication_url( $pledge_id, 'manage_pledge', get_permalink( get_page_by_path( 'manage-pledge' ) ), false )
+			// b/c false is set ^, the auth token can be used infinite # of times, but will still expire w/ in the 2 hour window
+		;
+
+		$message =
+			"Howdy $name, {$pledge->post_title} has created a Five for the Future pledge on WordPress.org and listed you as one of the contributors that they sponsor to contribute to the WordPress open source project. You can view their pledge at:" . "\n\n" .
+
+			get_permalink( $pledge_id ) . "\n\n" .
+
+			"To confirm that they're sponsoring your contributions, please review your pledges at:" . "\n\n" .
+
+			get_permalink( get_page_by_path( 'my-pledges' ) ) . "\n\n" .
+
+			"Please also update your WordPress.org profile to include the number of hours per week that you contribute, and the teams that you contribute to:\n\n" .
+
+			"https://profiles.wordpress.org/me/profile/edit/group/5/\n\n" .
+
+			"If {$pledge->post_title} isn't sponsoring your contributions, then you can ignore this email, and you won't be listed on their pledge."
+		;
+
+
+		$result = Email\send_email( $valid_admin_email, $subject, $message, $pledge_id );
+
+		if ( ! $result ) {
+			$result = new WP_Error( 'email_failed', 'Email failed to send' );
+		}
+	} else {
+		$error_message = sprintf(
+			__( 'That\'s not the address we have for this pledge, please try again. If you no longer know the correct address, please <a href="%s">email us</a>.', 'wporg-5ftf' ),
+			get_permalink( get_page_by_path( 'report' ) )
+		);
```

Comment:
Hm, I'm hesitant to give away too much info here. Somebody could try a bunch of email addresses until they find the right one, followed by having the ability to spam the pledge owner.

1) Could give a more generic message on both success and failures.
2) Captchas? :/

Reply:
🤔 

I can definitely see your point, but I also feel like generic error messages create a really bad UX, especially in a case like this where it's highly likely that admins will often forget which address they used.

We can definitely consider a CAPTCHA if abuse becomes a problem, but given our limited resources & timeframe, and the tiny amount of traffic this site gets, that doesn't seem like a high priority to me right now.